src/SimCore/GammaPhysics.cxx

```diff
@@ -9,60 +9,37 @@
 
 namespace simcore {
 
-// needed for GEANT4 10.3.0 and later
-//
-// TODO: I (Einar) don't think this really is necessary if we change
-// GetGammaProcessManager below
-#ifndef aParticleIterator
-#define aParticleIterator \
-  ((subInstanceManager.offset[g4vpcInstanceID])._aParticleIterator)
-#endif
+GammaPhysics::GammaPhysics(const G4String& name,
+                           const framework::config::Parameters& parameters)
+    : G4VPhysicsConstructor(name),
+      modelParameters{parameters.getParameter<framework::config::Parameters>(
+          "photonuclear_model")} {}
 
-G4ProcessManager* GammaPhysics::GetGammaProcessManager() const {
-  /**
-   * Not entirely clear that there isn't a simpler way to do this, but to keep
-   * this function extraction to a pure refactoring, I'm leaving it here for
-   * now. /Einar
-   */
-  aParticleIterator->reset();
+void GammaPhysics::ConstructParticle() {}
 
-  // Loop through all of the particles and find the "gamma".
-  while ((*aParticleIterator)()) {
-    auto particle{aParticleIterator->value()};
-    auto particleName{particle->GetParticleName()};
-    if (particleName == "gamma") {
-      return particle->GetProcessManager();
-    }
-  }
-  EXCEPTION_RAISE("GammaPhysics",
-                  "Was unable to access the process manager for photons, "
-                  "something is very wrong!");
-}
-
-void GammaPhysics::SetPhotonNuclearAsFirstProcess() const {
-  auto processManager{GetGammaProcessManager()};
-  // Get the process list associated with the gamma.
-  const auto processes{processManager->GetProcessList()};
-
-  for (int i{0}; i < processes->size(); ++i) {
-    const auto process{(*processes)[i]};
-    const auto processName{process->GetProcessName()};
-    if (processName == "photonNuclear") {
-      processManager->SetProcessOrderingToFirst(
-          process, G4ProcessVectorDoItIndex::idxAll);
-    }
-  }
-}
 void GammaPhysics::ConstructProcess() {
-  G4ProcessManager* processManager = GetGammaProcessManager();
-
+  G4ProcessManager* processManager = G4Gamma::Gamma()->GetProcessManager();
+  if (processManager == nullptr) {
+    EXCEPTION_RAISE("GammaPhysics",
+                    "Was unable to access the process manager for photons, "
+                    "something is very wrong!");
+  }
+  // configure our PN model based on runtime parameters
   auto pn = PhotoNuclearModel::Factory::get().make(
       modelParameters.getParameter<std::string>("class_name"),
       modelParameters.getParameter<std::string>("instance_name"),
       modelParameters);
   pn->removeExistingModel(processManager);
   pn->ConstructGammaProcess(processManager);
-  SetPhotonNuclearAsFirstProcess();
+  // put the PN process first in the ordering in case PN biasing is happening
```

Comment:
Do we have a reference in G4 or ldmx-sw for why the ordering matters? Could be useful to point people reading this to

Reply:
Unfortunately, all I have is a set of slides from 2010 that describe why process ordering matters.

https://indico.cern.ch/event/58317/contributions/2047449/attachments/992300/1411062/PartDef-ProcMan-AS.pdf

I'm not sure if its still even relevant _and_ it does not refer to the biasing framework as being affected. I only know this is necessary[^1] since I observed a difference while working on G4DarkBreM.

[^1]: "necessary" may even be too strong of a word. While working on the total cross section calculation, I was trying to debug why the process was never happening and so I tried putting it first in the ordering. I noticed that putting it first in the ordering meant that it was always being checked by Geant4 while Geant4 sometimes doesn't make it to the end of the ordering list.